src/demandlib/vdi/regions.py

```diff
@@ -0,0 +1,598 @@
+"""
+The region module combines profiles from VDI 4655 for heat and power demand of
+houses within one region.
+
+
+Module VDI 4655
+---------------
+This is an implementation of the calculation of economic efficiency
+using the annuity method defined in the German VDI 4655.
+
+    **VDI 4655**
+
+    **Reference load profiles of single-family and
+    multi-family houses for the use of CHP systems**
+
+    *May 2008 (ICS 91.140.01)*
+
+    *Verein Deutscher Ingenieure e.V.*
+
+    *VDI Standards Department*
+
+    *VDI-Platz 1, 40468 Duesseldorf, Germany*
+
+Reproduced with the kind permission of the Verein Deutscher Ingenieure e.V.
+
+Notes
+-----
+This script creates full year energy demand time series of domestic buildings
+for use in simulations. This is achieved by an implementation of the VDI 4655,
+which gives sample energy demands for a number of typical days ('Typtage').
+The energy demand contains heating, hot water and electricity.
+
+For a given year, the typical days can be matched to the actual calendar days,
+based on the following conditions:
+    - Season: summer, winter or transition
+    - Day: weekday or sunday (Or holiday, which counts as sunday)
+    - Cloud coverage: cloudy or not cloudy
+    - House type: single-family houses or multi-family houses (EFH or MFH)
+
+The holidays are loaded from an Excel file, the weather conditions are loaded
+from weather data (e.g. DWD TRY).
+Most of the settings for this script are controlled with a configuration file
+called 'config_file', the location of which is defined down below.
+
+SPDX-FileCopyrightText: Joris Zimmermann
+SPDX-FileCopyrightText: Uwe Krien
+
+SPDX-License-Identifier: MIT
+"""
+
+import calendar
+import datetime
+import os
+import warnings
+from collections import namedtuple
+
+import pandas as pd
+
+from demandlib.tools import add_weekdays2df
+from demandlib.vdi import dwd_try
+
+
+class Region:
+    def __init__(
+        self,
+        year,
+        try_region,
+        seasons=None,
+        holidays=None,
+        houses=None,
+        resample_rule=None,
```

Comment:
I am also adding the code snippets for changing the weather file.
```suggestion
        resample_rule=None,
        file_weather=None,
```